### PR TITLE
fix(i18n): update Chinese translation of JavaScript Algorithms and Da…

### PR DESCRIPTION
**Pull request by ZhichengChen**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

### CODE REVIEW

File: curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/caesars-cipher.chinese.md [hunk 1]
```diff
@@ -3,30 +3,37 @@ id: 56533eb9ac21ba0edf2244e2
 title: Caesars Cipher
 challengeType: 5
 isRequired: true
-videoUrl: ''
+forumTopicId: 16003
 localeTitle: 凯撒密码
 ---
 
 ## Description
-<section id="description">最简单和最广为人知的<dfn>密码之一</dfn>是<code>Caesar cipher</code> ，也称为<code>shift cipher</code> 。在<code>shift cipher</code>中，字母的含义被移动一些设定量。一种常见的现代用途是<a href="https://en.wikipedia.org/wiki/ROT13" target="_blank">ROT13</a>密码，其中字母的值移动了13个位置。因此&#39;A&#39;&#39;&#39;N&#39;，&#39;B&#39;&#39;&#39;O&#39;等等。编写一个函数，它将<a href="https://en.wikipedia.org/wiki/ROT13" target="_blank">ROT13</a>编码的字符串作为输入并返回一个已解码的字符串。所有字母都是大写的。不要转换任何非字母字符（即空格，标点符号），但要传递它们。如果卡住，请记得使用<a href="https://www.freecodecamp.org/forum/t/how-to-get-help-when-you-are-stuck-coding/19514" target="_blank">Read-Search-Ask</a> 。尝试配对程序。编写自己的代码。 </section>
```
Comment (Raman0716) [inline curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/caesars-cipher.chinese.md:11]: One of the simplest and most widely known ciphers is a Caesar cipher, also known as a shift cipher.
<section id="description">最简单和最广为人知的<dfn>密码之一</dfn>是<code>Caesar cipher</code> ，也称为<code>shift cipher</code> => <section id="description"><code>凯撒密码</code>是最简单和最广为人知的<dfn>密码之一</dfn>，也被称为<code>移位密码</code>

**Review by miyaliu666 — DISMISSED** [review]

File: curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/caesars-cipher.chinese.md [hunk 2]
```diff
@@ -3,29 +3,36 @@ id: 56533eb9ac21ba0edf2244e2
...
 ---
 
 ## Description
-<section id="description">最简单和最广为人知的<dfn>密码之一</dfn>是<code>Caesar cipher</code> ，也称为<code>shift cipher</code> 。在<code>shift cipher</code>中，字母的含义被移动一些设定量。一种常见的现代用途是<a href="https://en.wikipedia.org/wiki/ROT13" target="_blank">ROT13</a>密码，其中字母的值移动了13个位置。因此&#39;A&#39;&#39;&#39;N&#39;，&#39;B&#39;&#39;&#39;O&#39;等等。编写一个函数，它将<a href="https://en.wikipedia.org/wiki/ROT13" target="_blank">ROT13</a>编码的字符串作为输入并返回一个已解码的字符串。所有字母都是大写的。不要转换任何非字母字符（即空格，标点符号），但要传递它们。如果卡住，请记得使用<a href="https://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck-coding/19514" target="_blank">Read-Search-Ask</a> 。尝试配对程序。编写自己的代码。 </section>
+<section id='description'>
+<code>凯撒密码</code>是最简单和最广为人知的<dfn>密码之一</dfn>，也被称为<code>移位密码</code>。在<code>移位密码</code>中，明文中的字母通过按照一个固定数目进行偏移后被替换成新的字母。
+<a href="https://en.wikipedia.org/wiki/ROT13" target='_blank'>ROT13</a> 是一个被广泛使用的编码技术，明文中的所有字母都被移动 13 位。因此，'A' &harr; 'N', 'B' &harr;  'O' 等等。
+请编写一个函数，用于解码一个被 <a href="https://en.wikipedia.org/wiki/ROT13" target='_blank'>ROT13</a> 编码的字符串，然后返回解码后的结果。
+所有解码后的字母都必须为字母大写。请不要解码非字母的字符（例如，空格、标点符号），但你需要在结果中保留它们。
+如果你遇到了问题，请点击<a href='https://forum.freecodecamp.one/t/topic/157' target='_blank'>帮助</a>。
```
Comment (RandellDawson) [inline curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/caesars-cipher.chinese.md:16]: This line is not in the current English version.

File: curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/palindrome-checker.chinese.md [hunk 3]
```diff
@@ -3,47 +3,55 @@ id: aaa48de84e1ecc7c742e1124
...
 
 ## Description
-<section id="description">如果给定的字符串是回文，则返回<code>true</code> 。否则，返回<code>false</code> 。 <dfn>回文</dfn>是一个单词或句子，其拼写方式与前后相同，忽略标点符号，大小写和间距。 <strong>注意</strong> <br>您需要删除<strong>所有非字母数字字符</strong> （标点符号，空格和符号）并将所有内容转换为相同的大小写（小写或大写）以检查回文。我们会通过字符串具有不同的格式，如<code>&quot;racecar&quot;</code> ， <code>&quot;RaceCar&quot;</code>和<code>&quot;race CAR&quot;</code>等等。我们还将传递带有特殊符号的字符串，例如<code>&quot;2A3*3a2&quot;</code> ， <code>&quot;2A3 3a2&quot;</code>和<code>&quot;2_A3*3#A2&quot;</code> 。如果卡住，请记得使用<a href="https://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck-coding/19514" target="_blank">Read-Search-Ask</a> 。编写自己的代码。 </section>
+<section id='description'>
+如果给定的一个字符串是回文，那么返回<code>true</code>，否则返回<code>false</code>。
+<dfn>palindrome（回文）</dfn>，指在忽略标点符号、大小写和空格的前提下，正着读和反着读一模一样。
+<strong>注意：</strong><br>检查回文时，你需要先除去<strong>所有非字母数字的字符</strong>（标点、空格和符号）并且将所有字符转换成字母大写或字母小写。
+我们将会传入不同格式的字符串，例如：<code>"racecar"</code>、<code>"RaceCar"</code>、<code>"race CAR"</code>等等。
+我们也会传入一些包含特殊符号的字符串，例如<code>"2A3*3a2"</code>，<code>"2A3  3a2"</code>和<code>"2_A3*3#A2"</code>。
+如果你遇到了问题，请点击<a href='https://forum.freecodecamp.one/t/topic/157' target='_blank'>帮助</a>。
```
Comment (RandellDawson) [inline curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/palindrome-checker.chinese.md:17]: This line is not in the current English version.

File: curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/roman-numeral-converter.chinese.md [hunk 4]
```diff
@@ -3,73 +3,78 @@ id: a7f4d8f2483413a6ce226cac
...
+forumTopicId: 16044
 localeTitle: 罗马数字转换器
 ---
 
 ## Description
-<section id="description">将给定数字转换为罗马数字。所有<a href="http://www.mathsisfun.com/roman-numerals.html" target="_blank">罗马数字</a>答案都应以大写字母提供。如果卡住，请记得使用<a href="https://forum.freecodecamp.org/t/how-to-get-help-when-you-are-stuck-coding/19514" target="_blank">Read-Search-Ask</a> 。尝试配对程序。编写自己的代码。 </section>
+<section id='description'>
+把传入的数字转变为罗马数字。
+转换后的<a href="http://www.mathsisfun.com/roman-numerals.html" target="_blank">罗马数字</a>字母必须都是大写。
+如果你遇到了问题，请点击<a href='https://forum.freecodecamp.one/t/topic/157' target='_blank'>帮助</a>。
```
Comment (RandellDawson) [inline curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/roman-numeral-converter.chinese.md:14]: This line is not in the English version.

File: curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/telephone-number-validator.chinese.md [hunk 5]
```diff
@@ -3,75 +3,82 @@ id: aff0395860f5d3034dc0bfc9
...
+<section id='description'>
+如果传入的字符串是一个有效的美国电话号码格式，则返回<code>true</code>。
+只要是有效的美国电话号码的格式，用户可以按照他们的方式填写表单中的电话号码。以下是一些正确的例子（其他格式变形请参考以下例子）：
+<blockquote>555-555-5555<br>(555)555-5555<br>(555) 555-5555<br>555 555 5555<br>5555555555<br>1 555 555 5555</blockquote>
+在这个挑战中，你将会看到例如<code>800-692-7753</code>或者<code>8oo-six427676;laskdjf</code>的号码。你的任务是根据上面不同的格式组合，判断它是否美国号码。区号是必须的。如果提供国家代码，则必须确认国家代码为<code>1</code>。如果这是有效的美国电话就返回<code>true</code>，否则返回<code>false</code>。
+如果你遇到了问题，请点击<a href='https://forum.freecodecamp.one/t/topic/157' target='_blank'>帮助</a>。
```
Comment (RandellDawson) [inline curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/telephone-number-validator.chinese.md:16]: This line is not in the English version.

File: curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/cash-register.chinese.md [hunk 6]
```diff
@@ -3,33 +3,43 @@ id: aa2e6f85cab2ab736c9a9b24
...
+<section id='description'>
+编写一个用于收银机的函数<code>checkCashRegister()</code>，传入售价为第一个参数（<code>price</code>）、支付金额为第二个参数（<code>cash</code>）、收银机內的金额为第三个参数（<code>cid</code>）。
+<code>cid</code>是包含货币面值的二维数组。
+函数<code>checkCashRegister()</code>必须返回含有<code>status</code>键值和<code>change</code>键值的对象。
+如果收银机內的金额少于应找回的零钱数，或者你无法返回确切的数目时，返回<code>{status: "INSUFFICIENT_FUNDS", change: []}</code>。
+如果收银机內的金额等于应找回的零钱数，返回<code>{status: "CLOSED", change: [...]}</code>，其中<code>change</code>键值是收银机內的金额。
+否则，返回<code>{status: "OPEN", change: [...]}</code>，其中<code>change</code>键值是应找回的零钱数，并且它的面值由高到低排序。
+如果你遇到了问题，请点击<a href='https://forum.freecodecamp.one/t/topic/157' target='_blank'>帮助</a>。
```
Comment (RandellDawson) [inline curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/cash-register.chinese.md:18]: This line is not in the English version.

**Comment by miyaliu666** [comment]
@ZhichengChen Could you delete the last line in every project description? Thanks!

**Review by miyaliu666 — APPROVED** [review]

**Pull request merged**